modules/driver-mongo/src/test/kotlin/datamaintain/db/driver/mongo/test/AbstractMongoDbTest.kt

```diff
@@ -1,32 +1,142 @@
 package datamaintain.db.driver.mongo.test
 
 import com.mongodb.ConnectionString
+import com.mongodb.client.MongoClient
 import com.mongodb.client.MongoClients
+import com.mongodb.client.MongoCollection
+import com.mongodb.client.MongoDatabase
 import datamaintain.db.driver.mongo.MongoDriver
+import datamaintain.db.driver.mongo.MongoShell
 import org.bson.Document
+import org.junit.jupiter.api.AfterAll
 import org.junit.jupiter.api.AfterEach
-import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.params.provider.Arguments
+import org.testcontainers.containers.MongoDBContainer
 
 abstract class AbstractMongoDbTest {
-    val databaseName = "datamaintain-test"
-    val mongoHost = "localhost:27018"
-    val mongoUri = "mongodb://localhost:27018/datamaintain-test"
-    val connectionString = ConnectionString(mongoUri)
+    private val defaultDatabaseName = "datamaintain-test"
 
-    private val client = MongoClients.create(connectionString)
-    val database = client.getDatabase(databaseName)
+    // Fill when mongo container is up
+    private var currentMongoContainer: MongoDBContainer? = null
+    private var mongoUri: String? = null
+    private var connectionString: ConnectionString? = null
+    private var client: MongoClient? = null
+    private var database: MongoDatabase? = null
+    private var collection: MongoCollection<Document>? = null
 
-    val collection = database.getCollection(
-            MongoDriver.EXECUTED_SCRIPTS_COLLECTION, Document::class.java)
+    /**
+     * MongoUri of current mongo container
+     */
+    fun mongoUri(): String {
+        return mongoUri!!
+    }
 
-    @BeforeEach
-    fun init() {
-        cleanDb()
+    /**
+     * Database of current mongo container
+     */
+    fun database(): MongoDatabase {
+        return database!!
+    }
+
+    /**
+     * Collection of current mongo container
+     */
+    fun collection(): MongoCollection<Document> {
+        return collection!!
     }
 
     @AfterEach
-    fun cleanDb() {
-        collection.drop()
-        database.getCollection("simple").drop()
+    fun cleanDbConnection() {
+        if (client != null) {
+            client!!.close()
```

Comment:
There is no need for non null assertion here, Kotlin should know that the client is not null since you checked it just before